presto-raptor/src/test/java/com/facebook/presto/raptor/storage/organization/TestCompactionSetCreator.java

```diff
@@ -103,22 +111,52 @@ public void testNonTemporalRowCountBasedOrganizationSet()
     public void testTemporalCompactionNoCompactionAcrossDays()
             throws Exception
     {
-        long day1 = Duration.ofDays(Duration.ofNanos(System.nanoTime()).toDays()).toMillis();
-        long day2 = Duration.ofDays(Duration.ofMillis(day1).toDays() + 1).toMillis();
-        long day3 = Duration.ofDays(Duration.ofMillis(day1).toDays() + 2).toMillis();
+        long today = Duration.ofNanos(System.nanoTime()).toDays();
+        long startOfDay1 = LOCAL_TIME_ZONE.convertLocalToUTC(Duration.ofDays(today).toMillis(), false);
+        long startOfDay2 = LOCAL_TIME_ZONE.convertLocalToUTC(Duration.ofDays(today + 1).toMillis(), false);
+        long startOfDay3 = LOCAL_TIME_ZONE.convertLocalToUTC(Duration.ofDays(today + 2).toMillis(), false);
 
         List<ShardIndexInfo> inputShards = ImmutableList.of(
-                shardWithTemporalRange(TIMESTAMP, day1, day1),
-                shardWithTemporalRange(TIMESTAMP, day2, day2),
-                shardWithTemporalRange(TIMESTAMP, day2, day2),
-                shardWithTemporalRange(TIMESTAMP, day1, day1),
-                shardWithTemporalRange(TIMESTAMP, day3, day3));
+                shardWithTemporalRange(TIMESTAMP, startOfDay1, startOfDay2 - 100),
+                shardWithTemporalRange(TIMESTAMP, startOfDay1, startOfDay1),
+                shardWithTemporalRange(TIMESTAMP, startOfDay2, startOfDay3 - 100),
+                shardWithTemporalRange(TIMESTAMP, startOfDay2, startOfDay2 + 100),
+                shardWithTemporalRange(TIMESTAMP, startOfDay2, startOfDay2),
+                shardWithTemporalRange(TIMESTAMP, startOfDay3, startOfDay3));
 
         Set<OrganizationSet> actual = compactionSetCreator.createCompactionSets(temporalTableInfo, inputShards);
         assertEquals(actual.size(), 2);
 
         Set<OrganizationSet> expected = ImmutableSet.of(
-                new OrganizationSet(temporalTableInfo.getTableId(), extractIndexes(inputShards, 0, 3), OptionalInt.empty()),
+                new OrganizationSet(temporalTableInfo.getTableId(), extractIndexes(inputShards, 0, 1), OptionalInt.empty()),
+                new OrganizationSet(temporalTableInfo.getTableId(), extractIndexes(inputShards, 2, 3, 4), OptionalInt.empty()));
+        assertEquals(actual, expected);
+    }
+
+    @Test
+    public void testTemporalCompactionTimeZoneMigration()
```

Comment:
I'm inclined to say, let's leave this class alone entirely and add a new `TestTemporalFunction`. As long as the function is computing days correctly, this code will work.